### PR TITLE
Issue 1108: Fix state issues with Data Files Download button

### PR DESCRIPTION
**Pull request by jtwillis92**
### Summary of Changes
Closes issue #1108 

Resolves two issues with Data Files submissions that would result in the download button not displaying for newly uploaded files and unintended re-submissions of previously submitted Data Files. This is done through the introduction of a new Redux reducer `SET_FILE_SUBMITTED` which merges the API response of the created Data File with the app state.

### How to Test
_List the steps to test the PR_
These steps are generic, please adjust as necessary.
```
cd tdrs-frontend && docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d
cd tdrs-backend && docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d 
```

1. Open http://localhost:3000/ and sign in.
2. Navigate to the Data Files section as either OFA Admin or Data Analyst
3. Submit a Data File and confirm that the download button appears after the submission completes.
4. Submit another Data File without refreshing the page and confirm only 1 API call was made to `/v1/data_files/` and the previously submitted file is not submitted again.
5. Submit an infected file (EICAR.txt) and confirm an error is displayed. (There will be more work needed on this to show errors with better context but this at least gets them into the UI instead of no indication there was an error.)

### Deliverable 1: Accepted Features

> *Performance Standard(s): At the beginning of each sprint, the Product Owner and development team will collaborate to define a set of user stories to be completed during the sprint. Acceptance criteria for each story will also be defined. The development team will deliver code and functionality to satisfy these user stories.*

> *Acceptable Quality Level: Delivered code meets the acceptance criteria for each user story. Incomplete stories will be assessed and considered for inclusion in the next sprint.*

+ [x] Look up the acceptance criteria in the related issue; paste ACs below in checklist format.
+ [x] Check against the criteria:
  - [x] The download button appears immediately after successfully submitting sections in the Data Files UI.
  - [x] Further submissions without refreshing the page don't resend previously submitted Data Files.
  - [x] reviewed and tested by @ADPennington 

As Product Owner, @lfrohlich will decide if ACs are met.

### Deliverable 2: Tested Code

> *Performance Standard(s): Code delivered under the order must have substantial test code coverage. Version-controlled HHS GitHub repository of code that comprises products that will remain in the government domain.*

> *Acceptable Quality Level: Minimum of 90% test coverage of all code. All areas of code are meaningfully tested.*

+ Are all areas of code introduced in this PR meaningfully tested?
  + [ ] If this PR introduces backend code changes, are they meaningfully tested?
  + [x] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [x] Frontend coverage: ___ (see https://github.com/raft-tech/TANF-app README coverage dashboard)
  + [x] Backend coverage: ___ (see https://github.com/raft-tech/TANF-app README coverage dashboard)

### Deliverable 3: Properly Styled Code

> *Performance Standard(s): GSA 18F Front- End Guide*

> *Acceptable Quality Level: 0 linting errors and 0 warnings*

+ [x] Are backend code style checks passing on CircleCI?
+ [x] Are frontend code style checks passing on CircleCI?
+ [ ] Does this PR change any linting or CI settings?

### Deliverable 4: Accessible

> *Performance Standard(s): Web Content Accessibility Guidelines 2.1 AA standards*

> *Acceptable Quality Level: 0 errors reported using an automated scanner and 0 errors reported in manual testing*
+ [ ] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [ ] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with @iamjolly and @ttran-hub using Accessibility Insights reveal any errors introduced in this PR?
    + [See the full Accessibility Assessment plan here.](https://github.com/HHS/TANF-app/blob/main/docs/Technical-Documentation/how-government-will-test-a11y.md)

### Deliverable 5: Deployed

> *Performance Standard(s): Code must successfully build and deploy into the staging environment.*

> *Acceptable Quality Level: Successful deployment by assigning a 'Deploy with CircleCI - {env}' label*

+ [x] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### Deliverable 6: Documented

> *Performance Standard(s): Summary of user stories completed every two weeks. All dependencies are listed and the licenses are documented. Major functionality in the software/source code is documented, including system diagram. Individual methods are documented inline in a format that permits the use of tools such as JSDoc. All non-inherited 800-53 system security controls are documented in the Open Control or OSCAL format and HHS Section 508 Product Assessment Template (PAT) are updated as appropriate.*

> *Acceptable Quality Level: Combination of manual review and automated testing, if available*

+ [ ] If this PR introduces backend code, is that code documented both inline and overall?
+ [x] If this PR introduces frontend code, is that code documented both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?

### Deliverable 7: Secure

> *Performance Standard(s): Open Web Application Security Project (OWASP) Application Security Verification Standard 3.0*

> *Acceptable Quality Level: Code submitted must be free of medium- and high-level static and dynamic security vulnerabilities*

+ [x] Does the OWASP Scan pass on CircleCI?
+ [x] Do manual code review and manual testing detect any security issues?

### Deliverable 8: Context

> * Performance Standard(s): Code must be understandable and contextualized for the reviewers possess the knowledge and background necessary for analysis and constructive criticism to take place.

> * Acceptable Quality Level: Code submitted in the pull request has context.*
+ [x] Does this pull request contain sufficient inline comments providing relevant context for code snippets?
+ [x] Is the code understandable and lucid?
+ [x] Does this pull request provide background for why coding decisions were made?
+ [x] Can you as a reviewer explain and take ownership of these elements presented in this code review?

### CODE REVIEW

File: tdrs-frontend/src/actions/reports.js [hunk 1]
```diff
@@ -174,7 +175,7 @@ export const submit =
       )
     })
 
-    Promise.all(submissionRequests)
+    return Promise.all(submissionRequests)
```
Comment (jtwillis92) [inline tdrs-frontend/src/actions/reports.js:178]: This return doesn't really change the behavior, but was needed in order to assert asynchronous actions were recorded in tests.

File: tdrs-frontend/src/__mocks__/axios.js [hunk 2]
```diff
@@ -9,5 +9,10 @@ mockAxios.get = jest.fn(() =>
     data: {},
   })
 )
+mockAxios.post = jest.fn(() =>
```
Comment (jtwillis92) [inline tdrs-frontend/src/__mocks__/axios.js:12]: We need to declare this method in `__mocks__` to allow jest to mock POST requests over axios.

File: tdrs-frontend/src/actions/reports.js [hunk 3]
```diff
@@ -183,10 +184,17 @@ export const submit =
         })
         removeFileInputErrorState()
 
-        const submittedFiles = responses.map(
-          (response) =>
-            `${response?.data?.original_filename} (${response?.data?.extension})`
-        )
+        const submittedFiles = responses.reduce((result, response) => {
+          const submittedFile = response?.data
+          dispatch({
```
Comment (jtwillis92) [inline tdrs-frontend/src/actions/reports.js:189]: Each file needs to be marked as submitted individually, to increase performance `reduce` was used to allow one iteration through the responses.

**Review by abottoms-coder — APPROVED** [review]

**Review by jorgegonzalez — APPROVED** [review]
LGTM 👍

**Comment by ADPennington** [comment]
@jtwillis92 having some trouble testing this locally and on the deployed site (qasp). if locally, i continue to be re-directed back on the home page as if I hadnt logged in. If on deployed site, I get a `Request failed with status code 403` after clicking `submit data files`.

**Review by ADPennington — APPROVED** [review]
lgtm 🚀

**Pull request merged**